M/msolve/build_tarballs.jl

```diff
@@ -9,41 +9,40 @@ version = v"0.1.4"
 sources = [
     GitSource("https://gitlab.lip6.fr/safey/msolve.git", "be5a34d338d6ba66a5be2e67e93871b184789e87"),
     #= ArchiveSource("https://www.mathematik.uni-kl.de/~ederc/msolve-0.1.2.tar.gz", "ce6454b28477cb3b5670042faf7b3282e234fe1e0ee5a62c184d0512ef4126e1") =#
+    DirectorySource("./bundled"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir
 cd msolve/
 ./autogen.sh
-ac_cv_func_malloc_0_nonnull=yes ac_cv_func_realloc_0_nonnull=yes ./configure --with-gnu-ld --prefix=${prefix} --build=${MACHTYPE} --host=${target} --disable-openmp
+
+atomic_patch -p1 ${WORKSPACE}/srcdir/patches/sort_r.patch
+export CPPFLAGS="-I${includedir}"
+
+ac_cv_func_malloc_0_nonnull=yes ac_cv_func_realloc_0_nonnull=yes ./configure --with-gnu-ld --prefix=${prefix} --build=${MACHTYPE} --host=${target}
 make -j${nprocs}
 make install
 """
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
-platforms = [
-    Platform("i686", "linux"; libc = "glibc"),
-    Platform("x86_64", "linux"; libc = "glibc"),
-    Platform("aarch64", "linux"; libc = "glibc"),
-    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
-    Platform("powerpc64le", "linux"; libc = "glibc"),
-    Platform("x86_64", "macos"; )
-]
-
+platforms = supported_platforms(; experimental=true)
+filter!(!Sys.iswindows, platforms)  # no FLINT_jll available
 
 # The products that we will ensure are always built
 products = [
     LibraryProduct("libmsolve", :libmsolve),
-    LibraryProduct("libneogb", :libneogb)
+    LibraryProduct("libneogb", :libneogb),
 ]
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency(PackageSpec(name="GMP_jll", uuid="781609d7-10c4-51f6-84f2-b8444358ff6d"), v"6.1.2"; compat="6.1.2")
-    Dependency(PackageSpec(name="FLINT_jll"), compat = "~200.800")
-    Dependency(PackageSpec(name="MPFR_jll", uuid="3a97d323-0669-5f0c-9066-3539efd106a3"))
+    Dependency("GMP_jll", v"6.2.0"),
+    Dependency("FLINT_jll", compat = "~200.800.101"),
```

Comment:
Uhm, this is changing the compat with FLINT_jll, I believe the registry won't like this within a build version.  Is there a newer version of msolve to build to make this transition smoother?

Reply:
Yes, we can bump the version to 0.1.5 using msolve commit 42137aaa68ac6077d25f7cb8e72ff01ac6f14f41 .

@fingolfin This commit of msolve already includes the changes provided by your `sort_r.patch`, so I think we can remove the `atomic_patch` part, right?